csharp/Adapter/Microsoft.Spark.CSharp/Network/RioNative.cs

```diff
@@ -128,25 +129,34 @@ private void Init()
             var maxThreads = 1;
             if (useThreadPool)
             {
-                maxThreads = Environment.ProcessorCount;
+                maxThreads = (Environment.ProcessorCount + 1) >> 1;
```

Comment:
Just print out the processor number from Yarn cluster:
CSharpWorker [11500] is running on a machine with 32 processors

In Multithread worker mode, if YARN arrange multiple executors into one machine, it will start multiple CsharpWorker - so the total threads will be "processors X csharpworkers".